### PR TITLE
Deprecate `FixedLenStringArray`.

### PR DESCRIPTION
**Pull request by 1uc**

### CODE REVIEW

File: doc/migration_guide.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Migration Guide
+A collection of tips for migrating away from deprecated features.
+
+## Deprecation of `FixedLenStringArray`.
+The issue with `FixedLenStringArray` is that it is unable to avoid copies.
+Essentially, this class acts as a means to create a copy of the data in a
+format suitable for writing fixed-length strings; and the class acts as a tag
+for HighFive to overload on. The support of `std::string` in HighFive. Since
```
Comment (alkino) [inline doc/migration_guide.md:8]: rewrite a bit here?

This small sentence seems awkward.

File: include/highfive/H5DataType.hpp [hunk 2]
```diff
@@ -460,6 +461,11 @@ class FixedLenStringArray {
   private:
     vector_t datavec;
 };
+}  // namespace deprecated
+
+template <size_t N>
+using FixedLenStringArray H5_DEPRECATED("Use 'std::vector<string>'.") =
```
Comment (alkino) [inline include/highfive/H5DataType.hpp:467]:
```suggestion
using FixedLenStringArray H5_DEPRECATED("Use 'std::vector<std::string>'.") =
```

**Review by alkino — APPROVED** [review]

File: doc/migration_guide.md (hunk 1 above)
Comment (1uc) [reply doc/migration_guide.md]: Very, thanks for noticing. I've tried again.

File: include/highfive/H5DataType.hpp (hunk 2 above)
Comment (1uc) [reply include/highfive/H5DataType.hpp]: Thanks :)

**Pull request merged**